### PR TITLE
applet-window-buttons6: init at 0.13.0

### PR DESCRIPTION
**Pull request by A1ca7raz**
## Description of changes
Ported to Plasma 6.

And add A1ca7raz to maintainers.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by K900** [comment]
The giant pile of conditionals seems excessive. Maybe just split it into two expressions?

**Comment by A1ca7raz** [comment]
> The giant pile of conditionals seems excessive. Maybe just split it into two expressions?

I see. But how to split it into two expressions?

**Comment by K900** [comment]
Write a new expression for the qt6 version and keep the qt5 one as is?

**Comment by A1ca7raz** [comment]
> Write a new expression for the qt6 version and keep the qt5 one as is?

I know. But where I put this qt6 expression? Should I put this in another file such as qt6.nix, and callPackage qt6.nix in qt6-packages.nix?

**Comment by K900** [comment]
Maybe just pkgs/kde/misc.

File: pkgs/kde/third-party/applet-window-buttons/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    owner = "moodyhunter";
+    repo = "applet-window-buttons6";
+    rev = "v${version}";
+    hash = "sha256-S7JcDPo4QDqi/RtvreFNoPKwTg14bgaFGsuGSDxs5nM=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    extra-cmake-modules
+    wrapQtAppsHook
```
Comment (K900) [inline pkgs/kde/third-party/applet-window-buttons/default.nix:28]: Is wrapQtAppsHook actually necessary? Plasmoids should generally be just libraries.

File: pkgs/kde/third-party/applet-window-buttons/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    kcoreaddons
+    kdeclarative
+    kdecoration
+    libplasma
+  ];
+
+  meta = with lib; {
+    description = "Plasma 6 applet in order to show window buttons in your panels";
+    homepage = "https://github.com/moodyhunter/applet-window-buttons6";
+    license = licenses.gpl2Plus;
```
Comment (K900) [inline pkgs/kde/third-party/applet-window-buttons/default.nix:41]: I'm pretty sure it's `gpl2Only`?

File: pkgs/kde/third-party/applet-window-buttons/default.nix (hunk 1 above)
Comment (A1ca7raz) [reply pkgs/kde/third-party/applet-window-buttons/default.nix:28]: You are right.

**Pull request merged**